packages/webapp/src/components/Modals/QuickAssignModal/index.jsx

```diff
@@ -141,106 +98,19 @@ export default function TaskQuickAssignModal({
-        const hasWage = !!(amount || wageAtMoment);
-        shouldShowWageSection = !hasWage && !selectedWorker.doNotAskAgain;
-      }
-    }
-    setShowHourlyWageSection(shouldShowWageSection);
-  }, [user.is_admin, selectedWorker, wageAtMoment]);
-
-  useEffect(() => {
-    if (!user.is_admin) {
-      return;
-    }
-
-    resetField(HOURLY_WAGE_ACTION);
-    resetField(HOURLY_WAGE);
-  }, [user.is_admin, selectedWorker]);
-
-  useEffect(() => {
-    let shouldShowWageInput = false;
-
-    if (selectedHourlyWageAction) {
-      shouldShowWageInput = isYesOptionSelected(selectedHourlyWageAction);
-    }
-    if (!shouldShowWageInput) {
-      resetField(HOURLY_WAGE);
-    }
-
-    setShowHourlyWageInput(shouldShowWageInput);
-  }, [selectedHourlyWageAction]);
-
-  const radioOptions = [
-    {
-      label: t('ADD_TASK.HOURLY_WAGE.SET_HOURLY_WAGE'),
-      value: hourlyWageActions.SET_HOURLY_WAGE,
-    },
-    {
-      label: t('ADD_TASK.HOURLY_WAGE.FOR_THIS_TASK'),
-      value: hourlyWageActions.FOR_THIS_TASK,
-    },
-    {
-      label: t('common:NO'),
-      value: hourlyWageActions.NO,
-    },
-    {
-      label: t('ADD_TASK.HOURLY_WAGE.DONT_ASK'),
-      value: hourlyWageActions.DO_NOT_ASK_AGAIN,
-    },
-  ];
-
-  const renderHourlyWageSection = () => {
-    if (!showHourlyWageSection) {
-      return;
-    }
-
-    return (
-      <div className={styles.hourlyWageSection}>
-        <Label className={styles.warning} style={{ marginBottom: '24px' }}>
-          {t('ADD_TASK.HOURLY_WAGE.ASSIGNEE_WAGE_WARNING', { name: selectedWorker.label })}
-        </Label>
-        <Main style={{ marginBottom: '10px' }}>
-          {t('ADD_TASK.HOURLY_WAGE.WANT_TO_SET_HOURLY_WAGE')}
-        </Main>
-        <RadioGroup
-          hookFormControl={control}
-          name={HOURLY_WAGE_ACTION}
-          radios={radioOptions}
-          data-cy="quickAssign-hourlyWageAction"
-          style={{ marginBottom: '20px' }}
-        />
-        {showHourlyWageInput && (
-          <Input
-            unit={currencySymbol + t('ADD_TASK.HR')}
-            data-cy="quickAssign-hourlyWage"
-            label={t('WAGE.HOURLY_WAGE')}
-            step="0.01"
-            type="number"
-            onKeyPress={numberOnKeyDown}
-            hookFormRegister={register(HOURLY_WAGE, {
-              valueAsNumber: true,
-              min: { value: 0, message: t('WAGE.HOURLY_WAGE_RANGE_ERROR') },
-              max: { value: 999999999, message: t('WAGE.HOURLY_WAGE_RANGE_ERROR') },
-            })}
-            style={{ marginBottom: '24px' }}
-            errors={errors[HOURLY_WAGE] && (errors[HOURLY_WAGE].message || t('WAGE.ERROR'))}
-            toolTipContent={t('WAGE.HOURLY_WAGE_TOOLTIP')}
-            optional
-          />
-        )}
-      </div>
-    );
-  };
+  {
+    /*TODO: properly fix checkbox label overflow ST-272*/
+    /*TODO: LF-2932 - need to be able to unassign mutiple tasks at once */
+  }
+  const assignAllCheckbox = (
```

Comment:
useMemo(()=>{}, [selectedWorker.label])

Reply:
https://twitter.com/0xca0a
This guy always preaches not to declare components in the component body since components will always be recreated on each render.